Summary Table/Summary Table Dev Flextable.Rmd

```diff
@@ -0,0 +1,398 @@
+# LOCALITY <- "Stirling City with the Eastern Villages Bridge of Allan and Dunblane"
+# LOCALITY <- "Mid-Argyll, Kintyre and Islay"
+# LOCALITY <- "City of Dunfermline"
+# LOCALITY <- "Barra"
+# LOCALITY <- "Whalsay and Skerries"
+# LOCALITY <- "Inverclyde East"
+ 
+# setwd("~/list-localities-profiles/")
+
+## Project file path
+ lp_path <- "/conf/LIST_analytics/West Hub/02 - Scaled Up Work/RMarkdown/Locality Profiles/"
+ 
+# Source in global functions/themes script
+ source("./Master RMarkdown Document & Render Code/Global Script.R")
+ 
+# # demographics
+ source("./Demographics/1. Demographics - Population.R")
+ source("./Demographics/2. Demographics - SIMD.R")
+#
+# # housing
+ source("./Households/Households Code.R")
+#
+# # general health
+ source("./General Health/3. General Health Outputs.R")
+#
+# # lifestyle & risk factors
+ source("./Lifestyle & Risk Factors/2. Lifestyle & Risk Factors Outputs.R")
+#
+# # unscheduled care
+ source("./Unscheduled Care/2. Unscheduled Care outputs.R")
+
+library(tidyverse)
+library(knitr)
+library(gridExtra)
+library(grid)
+library(kableExtra)
+library(dplyr)
+library(flextable)
+library(officer)
+library(officedown)
+
+# detach(package:tidylog, unload = TRUE)
+```
+
+
+```{r echo=FALSE}
+# Set locality name object (with automatic wrapping)
+loc_name <- paste0(LOCALITY, "\nLocality") %>%
+  strwrap(width = case_when(n_loc < 4 ~ 20, n_loc == 4 ~ 15, n_loc > 4 ~ 8), simplify = FALSE)
+loc_name <- sapply(loc_name, paste, collapse = "\n")
+
+# Set other localities df
+otherloc_names <- paste0(other_locs$hscp_locality, "\nLocality") %>%
+  strwrap(width = case_when(n_loc < 4 ~ 20, n_loc == 4 ~ 15, n_loc > 4 ~ 8), simplify = FALSE)
+otherloc_names <- sapply(otherloc_names, paste, collapse = "\n")
+otherloc_names <- data.frame(loc_number = 2:n_loc, otherloc_names) %>%
+  pivot_wider(names_from = loc_number, values_from = otherloc_names)
+
+# Set HSCP name object
+hscp_name <- paste0(HSCP, "\nHSCP") %>%
+  strwrap(width = if_else(n_loc < 3, 30, 12), simplify = FALSE)
+hscp_name <- sapply(hscp_name, paste, collapse = "\n")
+
+summary_table <- data.frame(
+  Indicators = "Indicators",
+  Data_Type = "Data Type",
+  Time_Period = "Time Period",
+  Locality = loc_name,
+  Other_Localities = otherloc_names,
+  HSCP = hscp_name,
+  Scotland = "Scotland"
+)
+
+```
+
+## Demographics Summary Table
+
+
+```{r echo=FALSE}
+
+indicators <- c(
+  "Total population",
+  "Gender ratio male to female",
+  "Population over 65",
+  "Population in least deprived SIMD quintile",
+  "Population in most deprived SIMD quintile"
+)
+# List what data type they are
+datatype <- c("count", "ratio", "%", "%", "%")
+# Enter the time periods available
+time <- c(pop_max_year, pop_max_year, pop_max_year, "2020", "2020")
+
+loc_values <- c(total_population, paste0("1:", gender_ratio), over65, perc_top_quintile, perc_bottom_quintile)
+other_loc_values <- rbind(other_locs_total_pop, other_locs_gender_ratio, other_locs_over65, other_locs_simd)
```

Comment:
```suggestion
other_loc_values <- bind_rows(other_locs_total_pop, other_locs_gender_ratio, other_locs_over65, other_locs_simd)
```

Reply:
Just for info really (I looked this up!).... bind_rows can combine dataframes having different numbers of columns and assigns "NA" for missing column values whereas rbind throws an error.

Reply:
Is the error from rbind is desirable in this case?

I noticed that `bind_rows` was used instead of `rbind` later on, so either way there's a task to make it consistent 😆